app/models/mainstream_browse_page.rb

```diff
@@ -19,12 +19,16 @@ def active_top_level_browse_page
     linked_items("active_top_level_browse_page").first
   end
 
+  def ordered_second_level_browse_pages
+    details["ordered_second_level_browse_pages"]
+  end
+
   def second_level_browse_pages
     links = linked_items("second_level_browse_pages")
 
-    if second_level_pages_curated?
+    if second_level_pages_curated? and ordered_second_level_browse_pages
```

Comment:
Prefer to use `&&` instead of `and` (govuk-lint would've told you, but it's not running on this repo).

If you're worried about this being `nil` (perhaps unnecessarily?) you could also do:

```ruby
details["ordered_second_level_browse_pages"].to_a.index(link.content_id)
```